src/cosalib/build.py

```diff
@@ -96,7 +96,12 @@ def __init__(self, *args, **kwargs):
 
         self._found_files = {}
         self._workdir = kwargs.pop("workdir", os.getcwd())
-        self._tmpdir = tempfile.mkdtemp(prefix="build_tmpd")
+        tmpdir = os.path.join(self._workdir, "tmp")
+        os.environ['TMPDIR'] = tmpdir
```

Comment:
This isn't new, but... https://github.com/rust-lang/rust/pull/24741

AFAIK cosa isn't multi-threaded today, but it's just a time bomb to be mutating the environment.  We can fix as a follow-up.

Reply:
Yeah, we should probably just set it in the entrypoint instead?

Reply:
Hmm, I think it's trickier than that since the entrypoint has no notion of workdir. So yeah, let's do that as a follow-up!